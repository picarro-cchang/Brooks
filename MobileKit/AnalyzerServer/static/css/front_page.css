.horizontal-layout .container, 
.horizontal-layout .navbar-static-top .container, 
.horizontal-layout .navbar-fixed-top .container, 
/*.horizontal-layout .navbar-bottom .container, */
/*.horizontal-layout .navbar-bottom, */
.horizontal-layout .navbar-fixed-bottom .container {
    width:1350px
}

.vertical-layout .container, 
.vertical-layout .navbar-static-top .container, 
.vertical-layout .navbar-fixed-top .container, 
/*.vertical-layout .navbar-bottom, */
.vertical-layout .navbar-fixed-bottom .container {
    width:1070px
}

.main-body{
    margin-top:-2px;
    position: relative;
    height:90%;
}

.horizontal-layout #accordion-container{
    position: relative;
    width:940px;
    height:723px;
    z-index:10;
}
.horizontal-layout #map-container{
    position: absolute;
    width:400px;
    height:400px;
    right: 0;
    top: 0;
    z-index:10;
}
.horizontal-layout #ge-container{
    position: absolute;
    width:400px;
    height:400px;
    right: 0 ;   
    top: 350px;
    z-index:10;
}

.vertical-layout #accordion-container{
    position: relative;
    width:100%;
    height:400px;
    z-index:10;
}
.vertical-layout #map-container{
    position: absolute;
    width:49%;
    height:400px;
    left: 0;
    top: 410px;
    z-index:10;
}
.vertical-layout #ge-container{
    position: absolute;
    width:49%;
    height:400px;
    left: 51%;   
    top: 410px;
    z-index:10;
}

.navbar-bottom .container{
    background-color: #6eb43f;
    width: 100%;
    height: 5px;
    z-index:1;
}

.navbar-bottom{
    position: fixed;
    bottom: 0;
    height: 5px;
    /*left: 50%; */
    width:100%;
    z-index:1;
}

.horizontal-layout .navbar-bottom{
    /*margin-left:-685px;*/
}

.vertical-layout .navbar-bottom{
    /*margin-left:-535px;*/
}

.navbar-bottom .powered-container{
    position: absolute;
    bottom: 5px;
    right: 0;
}

.navbar-bottom .feedback-container{
    position: absolute;
    bottom: 15px;
    left: 0;
}

.layout-container{
    width:62px;
    height: 38px;
    position: absolute;
    top: 15px;
    right:10px;
    cursor: pointer;
}

.vertical-layout .layout-container{
    background:url("/static/images/icon_horizontal.png") repeat scroll 0 0 transparent;
}
.horizontal-layout .layout-container{
    background:url("/static/images/icon_vertical.png") repeat scroll 0 0 transparent;
}

#date-picker{
    left: 134px;
    position: absolute;
    top: 5px;
    width:112px;
}

#datepicker{
    width:100px;
}

#date-picker .ui-datepicker-trigger{
    right: 3px;
    position: absolute;
    top: 3px;
    cursor: pointer;
}

.layout-header{
    background-color: black;
    color: white;
    display: inline-block;
    font-size: 18px;
    height: 20px;
    padding-top: 10px;
    padding-bottom: 10px;
    width: 100%;
}

.layout-header span {
    margin-left: 10px;
}

.log-wrapper{
	cursor: pointer;
}

.log-wrapper.selected{
	background-color: #ddd;
}

.log-wrapper:hover{
	background-color: #efefef;
}

.mini-flot{
	width: 100%;
	height: 100px;
	display:none;
}

.log-wrapper .bar-wrapper{
    height: 20px;
    left: 283px;
    margin-bottom: -25px;
    position: relative;
    top: -34px;
    width: 46%;
}

.log-wrapper .btn{
    margin-right:5px;
}

.logo-container{
    background-color: white;
    height: 25px;
    left: 0;
    width: 127px;
    margin-top:23px;
}

.logo-container img{
	margin-left:20px;
}

.list-hr{
    margin-top: -10px;
    margin-bottom: 12px;
}

#map_container{
    width:100%;
    min-height: 300px;
    background-color: white;
}

#ge_container{
	width:100%;
	height: 300px;
	background-color: white;
}

#accordion2{
    display: inline-block;
    height: 90%;
    overflow-y: scroll;
    position: relative;
    width: 100%;
    background-color: white;
    /*border: 1px solid black;*/
}

#modal-chart{
    display: inline-block;
    width:450px;
    height: 200px;
}

#modal-selection{
    background-color: rgba(0, 0, 255, 0.5);
    bottom: 7px;
    display: inline-block;
    height: 40px;
    left: 26px;
    position: absolute;
    right: 8px;
    width: auto;
}

.navbar-inner{
    background-image: linear-gradient(to bottom, #FFFFFF, #FFFFFF);
    min-height: 70px;
    position: relative;
}


#live_view{
    top:8px;
    right: 40px;
    position: absolute;
}

.p3-icon-minus, .p3-icon-plus{
    display: inline-block;
    width: 20px;
    height: 20px;
    background: url("/static/images/arrow_right.png") no-repeat scroll 0 3px transparent;
}

.log-wrapper h5 .p3-icon-minus{
    display:none;
    background: url("/static/images/arrow_down.png") no-repeat scroll 0 4px transparent;
}

.log-wrapper h5.toggle-open .p3-icon-minus{
    display: inline-block;
}

.log-wrapper h5 .p3-icon-plus{
    display:inline-block;

}

.log-wrapper h5.toggle-open .p3-icon-plus{
    display: none;
}

.share-input{
    margin-left:19px;
    width:415px;
}

.log-content{
    height: 0;
    width:95%;
    display: inline-block;
    /*background-color: white;*/
    overflow: hidden;
    margin-left: 10px;
    margin-bottom: 10px;
}

#modal-value{
    background-color: white;
    border: 1px solid #999999;
    border-radius: 6px 6px 6px 6px;
    display: inline-block;
    font-size: 19px;
    font-weight: bold;
    height: 25px;
    left: 100%;
    padding-left: 10px;
    padding-top: 5px;
    position: absolute;
    text-align: left;
    top: 10px;
    width: 59px;
    margin-top:-15px;
}