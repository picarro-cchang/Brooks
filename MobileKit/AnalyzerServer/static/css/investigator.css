#pageContainer{
  display: inline-block;
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
  background-color: #ddd;
  z-index:1;
}

#pageContent{
  width:100%;
  top:0px;
  bottom:40px;
  height: 100%;
  background-color: white;
  position: absolute;
}

#pageFooter{
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 60px;
  z-index:5000;
  background-color: black;
}


#canvas_bubbles{
  position: absolute;
  z-index: 10;
  /*background-color: rgba(255,255,255,.3);*/
}
#canvas_wind{
  position: absolute;
  z-index: 11;
  /*background-color: rgba(255,255,255,.3);*/
}

#canvas_rose{
  position: absolute;
  top: 20px; 
  right: 20px;
  z-index:101;
  background: url(/static/images/rose.png);
  /*background-color:rgba(255,255,255,.3);*/
}

#canvas_rose_arrow{
  position: absolute;
  top: 0; 
  right: 0;
  z-index:100;

  /*background: url(/static/images/arrow.png);*/
  /*background-color:rgba(255,255,255,.3);*/
}

#canvas_rose_sprite{
  position: absolute;
  left: 100px; 
  right: 0;
  z-index:102;
  background: url(/static/images/arrow.png);
  /*background-color:rgba(255,255,255,.3);*/
}

#legend-controls{
  background-color: rgba(255,255,255,.9);
  display: inline-block;
  height: 100px;
  left: 4px;
  position: absolute;
  top: 91px;
  width: 48px;
  border: 1px solid #888888;
  border-radius: 5px 5px 5px 5px;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.4);
}

.control-header{
    background-color: #1C4A6C;
    height: 40px;
    padding: 5px;
    padding-left: 10px;
    width: 100%;
}

.control-header h1{
  color: white;
}

.control-close{
  position: absolute;
  right: 6px;
  top: 6px;
}

#pageFooter .icon{
  display: inline-block;
  height: 60px;
  width:64px;
  cursor: pointer;
  padding-left:10px;
  padding-right:10px;
  z-index: 5002;
}

#pageFooter .icon.selected{
  opacity: 1;
  background-color: #00ae43;
}

#pageFooter.grayscale .icon.selected{
  opacity: 1;
  background-color: #5C5C5C;
}

#pageFooter.grayscale{
   background-color: #333;
}

/*.icon.selected:hover{
  opacity: 1;
  background-color: #00ae43;
}

.icon:hover{
  opacity: 1;
  background-color: #666;
}*/

#pageFooter .icon-car{
  background: url(/static/images/icons/toolbar-car.png) no-repeat scroll 10px 5px transparent;
}

#pageFooter .icon-map{
  background: url(/static/images/icons/toolbar-map.png) no-repeat scroll 10px 5px transparent;
}

#pageFooter .icon-setting{
  background: url(/static/images/icons/toolbar-settings.png) no-repeat scroll 10px 5px transparent;
}

#pageFooter .icon-legend{
  background: url(/static/images/icons/toolbar-legend.png) no-repeat scroll 10px 5px transparent;
  opacity: .8;
}

#pageFooter .icon-chart{
  background: url(/static/images/icons/toolbar-chart.png) no-repeat scroll 10px 5px transparent;
}

#pageFooter .icon-iso{
  background: url(/static/images/icons/toolbar-iso.png) no-repeat scroll 10px 5px transparent;
}


#pageFooter .icon-logo{
  background: url(/static/images/public_url/picarro-inverted.png) no-repeat scroll 10px 9px transparent;
  width: 171px;
  padding-right:25px;
  position: fixed;
  right:0;
  bottom: -10px;
  z-index: -1;
}

#pageFooter .icon-pause{
  background: url(/static/images/icons/toolbar-pause-on.png) no-repeat scroll 10px 5px transparent;
}

#pageFooter .icon-pause.selected{
  background: url(/static/images/icons/toolbar-pause.png) no-repeat scroll 10px 5px #00ae43;
}

#pageFooter .icon-wind{
  background: url(/static/images/icons/toolbar-wind.png) no-repeat scroll 16px 5px transparent;
}

#pageFooter .icon-cursor{
  background: url(/static/images/icons/toolbar-cursor.png) no-repeat scroll 10px 5px transparent;
}

#pageFooter .icon-spacer{
  background-color: white;
  display: inline-block;
  width: 1px;
  height: 60px;
  margin-top: 0px;
  opacity: 0.5;
  margin-left:-4px;
  margin-right:-4px;
}


#pageHeader .nav_buttons{
  display: inline-block;
}

.nav_buttons_bottom{
  margin-top:7px;
  margin-left:7px;
}


/*settings*/

.settings-wrapper{
  display: inline-block;
  width: 100%;
  height: 80px;
  border-bottom: 1px solid #fff;
  padding:20px;
}

.settings-switch-wrapper{
  display: inline-block;
  width: 300px;
  height: 60px;
  right: 20px;
  margin-top: 10px;
  position: absolute;
  border: 1px solid #fff;
  border-radius: 30px 30px 30px 30px;
  box-shadow: inset 0px 0px 4px rgba(0,0,0,.2), 0px 0px 8px rgba(0,0,0,.1);
  background: -moz-linear-gradient(top,  #f2f2f2, #e1e1e1);
  background: -webkit-gradient(linear, left top, left bottom, from(#f2f2f2), to(#e1e1e1));
}

.switch_groove{
  display: inline-block;
  width: 100px;
  height: 43px;
  margin-left: 0px;
  margin-top: 8px;
  /*border: 1px solid #fff;*/
  border-radius: 22px 22px 22px 22px;
  box-shadow: inset 0px 0px 13px rgba(0,0,0,1), 0px 0px 3px rgba(255,255,255,.1);
  background: -moz-linear-gradient(top,  #313131, #4a4a4a);
  background: -webkit-gradient(linear, left top, left bottom, from(#313131), to(#4a4a4a));
  cursor:pointer;
}

.settings-wrapper.selected .switch_handle{
  left:60px;
}
.switch_handle{
  display: inline-block;
  position: relative;
  width: 40px;
  height: 40px;
  top:2px;
  left:2px;
  background: url("/static/images/switch_marker.png") no-repeat scroll 0 0 transparent;
}

.switch_text{
    display: inline-block;
    margin: 7px;
    position: relative;
    text-align: center;
    top: -12px;
    width: 80px;
    font-weight: bold;
    font-size: 16px;   
}

.settings-wrapper .label-text{
    color: #334D70;
    font-size: 27px;
    font-weight: bold;
    height: 80px;
    margin-top: 30px;
    position: absolute;
    text-align: right;
    width: 300px;
}

.settings-wrapper .right_text{
  opacity: .5;
}
.settings-wrapper .left_text{
  opacity:1;
}

.settings-wrapper.selected .right_text {
  opacity:1;
}
.settings-wrapper.selected .left_text {
  opacity: .5;
}

/*controls*/


#map_controls{
  width:100%;
  left:0%;
  height: 100%;
  top:0;
  position: absolute;
}

#setting_controls{
  width:100%;
  left:0%;
  height: 100%;
  top:0;
  position: absolute;
  display: inline-block;
  background-color: gray;
  z-index: 4001;
  display: none;
  margin: auto;
  background: #e0e0e0;
}

#car_controls, #capture_controls{
  position: absolute;
  width: 168px;
  height: 128px;
  left: 0px;
  bottom: -130px;
  z-index: 4999;
}

#capture_controls{
  left:674px;
  height:130px;
}

#capture_controls .icon{
  border-bottom: 1px solid white;
}

#car_controls .icon, #capture_controls .icon{
  background-color: black;
  height: 64px;
  width: 84px;
  display: inline-block;
  color: white;
  font-weight: bold;
  font-size: 16px;
  text-align: center;
  cursor: pointer;
  line-height:4em
}

#car_controls .icon.selected, #capture_controls .icon.selected{
  background-color: #00ae43;
}

.car-signal{
  cursor: pointer;
  opacity:.8;
}


.car-signal:hover{
  opacity:1;
}

#car_controls_inner #car_signal-top{
    background: url("/static/images/signal-top-off.png") no-repeat scroll 0 0 transparent;
    height: 199px;
    left: 283px;
    position: absolute;
    top: -27px;
    width: 209px;
}

#car_controls_inner #car_signal-top.active{
  background: url("/static/images/signal-top-on.png") no-repeat scroll 0 0 transparent;
  opacity:1;
}

#car_controls_inner #car_signal-front{
    background: url("/static/images/signal-top-off.png") no-repeat scroll 0 0 transparent;
    height: 199px;
    left: -2px;
    position: absolute;
    top: 240px;
    width: 104px;
}

#car_controls_inner #car_signal-front.active{
  background: url("/static/images/signal-top-on.png") no-repeat scroll 0 0 transparent;
  opacity:1;
}


.time-zone{
  margin-top:40px;
  margin-left:20px;
}


#map2_canvas{
  position: absolute;
  height:auto;
  width: 100%;
  height:80%;
  top:0;
  background-color: white;
}

#concPlot2{
  display: none;
}


#debugoutput{
  position: absolute;
  left: 5px;  
  top: 100px;
  height: 400px;
  width: 200px;
  overflow: scroll;
}

/*
.generic_slider{
	width:400px;
	margin-top:14px;
	display: inline-block;
	margin-right: 20px;
}*/

#time_slider{
    height: 90%;
    left: 60px;
    position: absolute;
    top: 5%;
}
#range_slider{
    height: 90%;
    left: 107px;
    position: absolute;
    top: 5%;
}

#map2_flot{
    background-color: #eee;
    height: 20%;
    left: 0;
    position: absolute;
    bottom: 0;
    width: 80%;
    z-index: 3000;
}
#isotopic_layer{
  width:100%;
  height:100%;
  bottom:0;
  position: absolute;
}

.iso-display{
  position: absolute;
  display: inline-block;
  background-color: black;
  height:auto;
  bottom:8px;
  top:8px;
  opacity:.16;
}

#map_realtime_flot{
    background-color: #eee;
    height: 20%;
    right: 4%;
    position: absolute;
    bottom: 0;
    width: 16%;
    z-index: 3000;
}
#realtime_display{
    background-color: #eeeeee;
    height: 20%;
    right: 0;
    position: absolute;
    bottom: 0;
    width: 4%;
    z-index: 3000;

}
#realtime_num{
    background: none repeat scroll 0 0 white;
    border: 1px solid gray;
    bottom: -10%;
    font-size: 14px;
    font-weight: bold;
    height: 20px;
    margin-bottom: -10px;
    margin-left: -2px;
    padding: 2px;
    position: absolute;
    width: 84%;
    border-radius: 5px;
    text-align: center;
}
#map2_histo{
    background-color: #eee;
    height: 100px;
    left: 200px;
    position: absolute;
    top: 400px;
    width: 600px;
}

#legend-controls .legend-wrapper{
    display:inline-block;
    width:40px;
    height:40px;
    position:absolute;
    cursor: pointer;
}

.legend-circle {
    border-radius: 50%;
    background-color: red;
    display: inline-block;
    position: absolute;
    bottom: 5px;
}

.legend-label {
    bottom: 0;
    position: absolute;
    left: 5px;
}

#id_legend_canvas{
    position: absolute;
    width: 1000px;
    height: 40px;
}

.blue-circle{
    z-index:1000;
}

.red-circle{
    z-index:2000;
}

#color-picker{
    position:absolute;
    height: 436px;
    width: 180px;
    margin-top: 40px;
}

#color-picker .color-wrapper{
    display: inline-block;
    cursor: pointer;
}

#color-picker .color-inner{
    display: inline-block;
    width: 20px;
    height: 20px;
}



