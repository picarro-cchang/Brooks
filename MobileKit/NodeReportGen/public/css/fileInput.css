div.custom_file_upload {
    width: 500px;
    height: 20px;
    margin-bottom: 19px;
}

input.file {
    width: 300px;
    height: 20px;
    border: 1px solid #BBB;
    border-right: 0;
    color: #888;
    padding: 5px;
    
    -webkit-border-top-left-radius: 5px;
    -webkit-border-bottom-left-radius: 5px;
    -moz-border-radius-topleft: 5px;
    -moz-border-radius-bottomleft: 5px;
    border-top-left-radius: 5px;
    border-bottom-left-radius: 5px;
    
    outline: none;
}

div.file_upload {
    width: 200px;
    height: 24px;
    
    /*
    background: #7abcff;
    background: -moz-linear-gradient(top,  #7abcff 0%, #60abf8 44%, #4096ee 100%);
    background: -webkit-gradient(linear, left top, left bottom, color-stop(0%,#7abcff), color-stop(44%,#60abf8), color-stop(100%,#4096ee));
    background: -webkit-linear-gradient(top,  #7abcff 0%,#60abf8 44%,#4096ee 100%);
    background: -o-linear-gradient(top,  #7abcff 0%,#60abf8 44%,#4096ee 100%);
    background: -ms-linear-gradient(top,  #7abcff 0%,#60abf8 44%,#4096ee 100%);
    background: linear-gradient(top,  #7abcff 0%,#60abf8 44%,#4096ee 100%);
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#7abcff', endColorstr='#4096ee',GradientType=0 );
    */
    display: inline;
    position: absolute;
    overflow: hidden;
    cursor: pointer;
    
    -webkit-border-top-right-radius: 5px;
    -webkit-border-bottom-right-radius: 5px;
    -moz-border-radius-topright: 5px;
    -moz-border-radius-bottomright: 5px;
    border-top-right-radius: 5px;
    border-bottom-right-radius: 5px;
    
    font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
    font-size: 15px;
    color: #FFF;
    text-align: center;
    padding-top: 8px;
}

div.file_upload:before {
    content: 'Browse';
    position: absolute;
    left: 0; right: 0;
    text-align: center;    
    cursor: pointer;
}

div.file_upload input {
    position: relative;
    height: 24px;
    width: 200px;
    display: inline;
    cursor: pointer;
    opacity: 0;
}