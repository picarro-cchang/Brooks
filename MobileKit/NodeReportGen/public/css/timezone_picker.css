.timezone-picker fieldset {
  width: 600px;
  margin: 10px auto;
}

.fieldset-legend {
  font-size: 70%; 
}

.timezone-picker label {
  font-weight: bold;
}

#id_timezoneModal {
  width: 650px;
  margin: -250px 0 0 -325px;
}

#id_timezoneModal .modal-body {
  max-height: 800px;
  overflow: hidden;
}

#wrapper {
  overflow: hidden;
}

#page {
  text-align: left;
  max-width: 800px;
  margin: auto;
}

#resize-form {
  text-align: center;
}

#timezone-picker {
  text-align: center;
  margin: 10px -240px;
}

#timezone-picker div,
#timezone-picker map {
  margin: auto;
}

#content {

}

#footer {
  font-size: 90%;
  margin: 20px auto;
  text-align: center;
}
