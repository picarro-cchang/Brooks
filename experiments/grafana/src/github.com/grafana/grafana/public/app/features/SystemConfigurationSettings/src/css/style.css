
body {
    margin: 0;
}

header > img {
    height: 80%;
    margin-left: 10%;
}

.column {
  float: left;
  padding: 5px;
}
/*.left-column {
  -webkit-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
-moz-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
}*/

.right-column {
  width: 800px;
}

.row:after {
  content: "";
  display: table;
  clear: both;
}


.fancyShadowAround{
-webkit-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
-moz-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
  
}

.InstrumentViewWraper{
  height:98%;
  top:10px;
  bottom:10px;
  right:10px;
  left:0px;
-webkit-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
-moz-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
}

.InstrumentsListWraper{
  top:10px;
  bottom:10px;
  left:0px;
-webkit-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
-moz-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
}

.RackPositionsComponent{
  margin-left: auto;
  margin-right: auto;
  width: 90%;
}

.NewInstrumentsComponent{
    margin-left: auto;
  margin-right: auto;
  width: 90%;
}

.InstrumentControls{
  padding: 25px 20px 25px 40px;
  margin-top: 20px;
  height:98%;
-webkit-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
-moz-box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
box-shadow: 0px 0px 9px 4px rgba(0,0,0,0.29);
}


.instrumentControls-header {
  padding: 20px;
  text-align: left;
}

.instrumentControls-column{
  float: left;
  width: 33.33%;
  padding: 15px;
}

.row:after {
  content: "";
  display: table;
  clear: both;
}

.warning-timestamp{
}


.footer {
  background-color: #F1F1F1;
  text-align: center;
  padding: 10px;
}
/*
.instrumentControls-right-side{
  float: right;
  padding: 5px;
}*/


.container_row{
  display: grid;
}

.layer1, .layer2{
  grid-column: 1;
  grid-row: 1;
}


.active-instrument{
  border: 1px solid rgba(255, 0, 0, .5);
}
.inactive-instrument{
  border: 1px solid rgba(255, 0, 0, 0);
}
/*.layer1 span{
  color: #fff;
  background: #000cf6;
}

.layer2{
  background: rgba(255, 0, 0, 0.4);
}
*/

.instrument-widget {
    display: flex;
/*
    justify-content: flex-start;*/
    /*align-items: center;*/
    width:95%;
    padding: 30px 20px 0;
    border: 1px solid #efefef;
    border-bottom: 1px solid #cecece;
    font-family: Roboto, sans-serif;
    font-weight: 100;
    font-size: 15px;
    color: #333333;
}


.positions-message.accepted.show{
  color:green;
  transition:color 200ms ease-in;
}


.positions-message.error.show{
  color:red;
  transition:color 200ms ease-in;
}







.color-box {
    width: 10px;
    height: 10px;
    display: inline-block;
    background-color: #ccc;
}















header > p {
    font-family: VT323, monospace;
    color: whitesmoke;
    font-size: 50px;
    margin-left: 60px;
}

.new-instrument-widget {
    float: right;
    margin: auto;
    width: 540px;
    display: flex;
    flex-direction: column;
    align-items: center;
    border: 1px solid #efefef;
}


.rack-instrument-widget {
    float: left;
    margin: auto;
    /*width: 90%;*/
    width:500px;
    display: flex;
    flex-direction: column;
    align-items: center;
    border: 1px solid #efefef;
}


.control-component-widget {
    float: right;
    margin: auto;
    /*width: 90%;*/
    width:540px;
    display: flex;
    flex-direction: column;
    align-items: center;
    border: 1px solid #efefef;
    padding: 10px;
}


.button5 {
  background-color: white;
  color: black;
  border: 2px solid #555555;
  box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);
}

.button5:hover {
  background-color: #555555;
  color: white;
}

.parentContainer{
    background-color: black;
    min-width: 1100px;
}

.rack-instrument-widget {
    top: 0px;
    left: 5%;
    width: 505px;
    position: absolute;
}


.example-component {
  width: 100px;
  height: 100px;
  background-color: red;
  -webkit-transition: 0.5s;
  -moz-transition: 0.5s;
  -o-transition: 0.5s;
  transition: 0.5s;
  opacity: 0;
  visibility: hidden; 
}
.example-component.show {
  opacity: 1;
  visibility: visible; 
}

.Petr {
  cursor: default;
}