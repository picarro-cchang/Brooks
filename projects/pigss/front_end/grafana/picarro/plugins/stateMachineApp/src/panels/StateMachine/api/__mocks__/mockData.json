{
    "standby": "ACTIVE",
    "identify": "READY",
    "loop_plan": "READY",
    "run": "READY",
    "plan": "READY",
    "plan_run": "READY",
    "plan_loop": "READY",
    "reference": "READY",
    "run_plan": "READY",
    "edit": "READY",
    "clean": {
      "1": "READY",
      "3": "READY",
      "4": "READY"
    },
    "bank": {
      "1": "READY",
      "3": "READY",
      "4": "READY"
    },
    "channel": {
      "1": {
        "1": "DISABLED",
        "2": "AVAILABLE",
        "3": "DISABLED",
        "4": "AVAILABLE",
        "5": "DISABLED",
        "6": "AVAILABLE",
        "7": "DISABLED",
        "8": "AVAILABLE"
      },
      "3": {
        "1": "DISABLED",
        "2": "AVAILABLE",
        "3": "DISABLED",
        "4": "AVAILABLE",
        "5": "DISABLED",
        "6": "AVAILABLE",
        "7": "DISABLED",
        "8": "AVAILABLE"
      },
      "4": {
        "1": "DISABLED",
        "2": "AVAILABLE",
        "3": "DISABLED",
        "4": "AVAILABLE",
        "5": "DISABLED",
        "6": "AVAILABLE",
        "7": "DISABLED",
        "8": "AVAILABLE"
      }
    }
}
