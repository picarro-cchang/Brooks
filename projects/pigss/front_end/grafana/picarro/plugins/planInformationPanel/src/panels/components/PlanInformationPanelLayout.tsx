import React, { Component } from "react";
import { PlanInformationPanel } from "./PlanInformationPanel";
import deepmerge from "deepmerge";
import { Plan } from "../types";
import API from "./../api/API";
import "./planInformation.css";

interface State {
  uistatus: { [key: string]: string };
  plan: Plan;
  initialized: boolean;
  timer: number;
  ws: WebSocket;
}

const REFRESH_INTERVAL = 5;
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;
export class PlanInformationPanelLayout extends Component<any, any> {
  constructor(props) {
    super(props);
    this.state = {
      ws: null,
      timer: null,
      initialized: false,
      uistatus: {},
      plan: {
        max_steps: 32,
        panel_to_show: 0,
        current_step: 1,
        focus: { row: 0, column: 0 },
        last_step: 0,
        steps: {},
        num_plan_files: 0,
        plan_files: {},
        plan_filename: "",
        bank_names: {
          1: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: "",
            },
          },
          2: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: "",
            },
          },
          3: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: "",
            },
          },
          4: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: "",
            },
          },
        },
      },
    };
  }

  timeout = 250;
  ws = new WebSocket(socketURL);
  handleData(o: any) {
    if (this.state.initialized) {
      if ("uistatus" in o) {
        const uistatus = deepmerge(this.state.uistatus, o.uistatus);
        if (o.uistatus.timer) {
          this.setState({ timer: o.uistatus.timer });
        }

        this.setState({ uistatus });
      } else if ("plan" in o) {
        const plan = deepmerge(this.state.plan, o.plan);
        this.setState({ plan });
        const current_step = this.state.plan.current_step;
      }
    }
  }

  getDataViaApi = () => {
    const uiStatusData = API.getRequest(`http://${apiLoc}/uistatus`).then(
      (response) => {
        response.json().then((obj) => {
          this.setState(deepmerge(this.state, { uistatus: obj }));
        });
      }
    );
    const planData = API.getRequest(`http://${apiLoc}/plan`).then(
      (response) => {
        response.json().then((obj) => {
          this.setState(deepmerge(this.state, { plan: obj }));
        });
      }
    );
    Promise.all([uiStatusData, planData]).then(() => {
      this.setState(deepmerge(this.state, { initialized: true }));
    });
  };

  componentDidMount() {
    this.getDataViaApi();
    this.connect();
  }

  componentWillUnmount() {
    console.log("Component will unmount");
    if (this.ws && this.ws.readyState === 1) {
      this.ws.send("CLOSE");
      this.ws.close(1000, "Client Initited Connection Termination");
    }
  }

  connect = () => {
    const that = this;
    let connectInterval;

    this.ws.onopen = () => {
      this.setState({ ws: this.ws });
      that.timeout = 250;
      clearTimeout(connectInterval);
    };

    this.ws.onmessage = (evt) => {
      const message = JSON.parse(evt.data);
      this.handleData(message);
    };

    this.ws.onclose = (e) => {
      console.log(
        `Socket is closed. Reconnect will be attempted in ${Math.min(
          10000 / 1000,
          (that.timeout + that.timeout) / 1000
        )} second.`,
        e.reason
      );
      that.timeout = that.timeout + that.timeout; // increment retry interval
      connectInterval = setTimeout(this.check, Math.min(10000, that.timeout)); // call check function after timeout
    };
  };

  check = () => {
    const { ws } = this.state;
    if (!ws || ws.readyState == WebSocket.CLOSED) this.connect(); // check if websocket instance is closed, if so call `connect` function.
  };

  render() {
    return (
      <div id="outer">
        {this.state.initialized ? (
          <PlanInformationPanel
            uistatus={this.state.uistatus}
            plan={this.state.plan}
            timer={this.state.timer}
          />
        ) : null}
      </div>
    );
  }
}
