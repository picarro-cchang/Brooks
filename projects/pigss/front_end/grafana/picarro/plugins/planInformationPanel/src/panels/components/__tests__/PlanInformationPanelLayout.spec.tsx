import React from "react";
import { shallow, mount } from "enzyme";
import "jest-styled-components";
import WS from "jest-websocket-mock";
import { PlanInformationPanelLayout } from "./../PlanInformationPanelLayout";
import "jest-fetch-mock";
import "react-toastify/dist/ReactToastify.css";
import mockData from "./../../api/__mocks__/mockData.json";
import mockPlan from "./../../api/__mocks__/mockPlan.json";

jest.mock("./../../api/API.ts");

const socketURL = `ws://localhost:1234`;
const mockWS = jest.fn(element => {
  return element;
});
const server = new WS(socketURL, { jsonProtocol: true });
const client = new WebSocket(socketURL);

describe("<PlanInformationPanelLayout />", () => {
  const handleData = jest.spyOn(PlanInformationPanelLayout.prototype, "handleData");
  const wrapper = mount(<PlanInformationPanelLayout />);
  const instance = wrapper.instance() as PlanInformationPanelLayout;

  it("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Websocket Send Message", async () => {
    await server.connected;
    server.send({"uistatus": {"timer": 0}})
  })
 
});
