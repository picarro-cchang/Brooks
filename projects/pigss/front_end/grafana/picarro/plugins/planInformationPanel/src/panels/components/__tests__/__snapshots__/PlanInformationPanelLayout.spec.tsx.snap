// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanInformationPanelLayout /> Snapshot 1`] = `
<PlanInformationPanelLayout>
  <div
    id="outer"
  />
</PlanInformationPanelLayout>
`;
